llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -6472,12 +6472,21 @@ def : Pat<(i32 (int_aarch64_neon_uaddlv (v8i16 (AArch64uaddlp (v16i8 V128:$op)))
             (v8i16 (SUBREG_TO_REG (i64 0), (UADDLVv16i8v V128:$op), hsub)),
             ssub))>;
 
+def : Pat<(v4i32 (AArch64uaddlv (v8i16 (AArch64uaddlp (v16i8 V128:$op))))),
```

Comment:
Does there need to be a v8i8->v4i16->v4i32 version of this pattern too?

Reply:
Yep, let me add the pattern.